services/order.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from db.models import Order, Ticket
+
+
+@transaction.atomic
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: Optional = None) -> Order:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
+    if date:
+        order.created_at = date
+        order.save()
```

Comment:
Manually setting the `created_at` field can lead to inconsistencies, as it overrides the automatic timestamping provided by `auto_now_add=True`. Consider removing this logic unless there's a specific requirement to allow custom dates.